### PR TITLE
Update number_input.dart

### PR DESCRIPTION
**Pull request by KushagraVasal**
number_input min max is buggy; here is the fix.

### CODE REVIEW

**Comment by KushagraVasal** [comment]
Review the code before merging

**Comment by KushagraVasal** [comment]
And also please comment on #47

File: lib/src/components/form/number_input.dart [hunk 1]
```diff
@@ -154,16 +154,17 @@ class _NumberInputState extends State<NumberInput> {
...
-                            _lastValidValue > widget.min!) {
+                        if (widget.max == null ||
+                            _lastValidValue < widget.max!) {
                           double oldValue = _value.toDouble();
                           _lastValidValue = oldValue + widget.step;
                           _controller.text = widget.allowDecimals
                               ? _lastValidValue.toString()
                               : _lastValidValue.toInt().toString();
                           widget.onChanged?.call(_lastValidValue);
+                          setState(() {});
```
Comment (sunarya-thito) [inline lib/src/components/form/number_input.dart:167]: Is rebuilding the widget necessary here?

File: lib/src/components/form/number_input.dart [hunk 2]
```diff
@@ -186,16 +187,17 @@ class _NumberInputState extends State<NumberInput> {
...
-                            _lastValidValue < widget.max!) {
+                        if (widget.min == null ||
+                            _lastValidValue > widget.min!) {
                           double oldValue = _value.toDouble();
                           _lastValidValue = oldValue - widget.step;
                           _controller.text = widget.allowDecimals
                               ? _lastValidValue.toString()
                               : _lastValidValue.toInt().toString();
                           widget.onChanged?.call(_lastValidValue);
+                          setState(() {});
```
Comment (sunarya-thito) [inline lib/src/components/form/number_input.dart:200]: Refer to previous comment.

**Review by sunarya-thito — CHANGES_REQUESTED** [review]

File: lib/src/components/form/number_input.dart (hunk 1 above)
Comment (KushagraVasal) [reply lib/src/components/form/number_input.dart:167]: disabled button not enabling and enabled button disabling without rebuilding

File: lib/src/components/form/number_input.dart (hunk 2 above)
Comment (KushagraVasal) [reply lib/src/components/form/number_input.dart:200]: same as above

**Review by sunarya-thito — APPROVED** [review]

**Comment by sunarya-thito** [comment]
LGTM

**Pull request merged**